QuizApp/LoginView/LoginViewController.swift

```diff
@@ -11,47 +11,90 @@ import SnapKit
 
 
 class LoginViewController: UIViewController {
-
-//    var dataService = DataService()
+    
+    //    var dataService = DataService()
     private var presenter: LoginPresenter!
-
+    
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet weak var emailField: UITextField!
     @IBOutlet weak var passwordField: UITextField!
     @IBOutlet weak var errorLable: UILabel!
     @IBOutlet weak var loginButton: UIButton!
-
+    
+    
+    
     convenience init(presenter: LoginPresenter) {
         self.init()
         self.presenter = presenter
         self.presenter.setDelegate(delegate: self)
     }
-
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         self.title = "PopQuiz"
-
+        
         addConstraints()
         errorLable.isHidden = true
         emailField.attributedPlaceholder = NSAttributedString(string: "Email",
-                attributes:
-                [NSAttributedString.Key.foregroundColor: UIColor.white])
+                                                              attributes:
+                                                                [NSAttributedString.Key.foregroundColor: UIColor.white])
         passwordField.attributedPlaceholder = NSAttributedString(string: "password",
-                attributes:
-                [NSAttributedString.Key.foregroundColor: UIColor.white])
-
+                                                                 attributes:
+                                                                    [NSAttributedString.Key.foregroundColor: UIColor.white])
+        
     }
-
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        titleLabel.transform = titleLabel.transform.scaledBy(x: 0, y: 0)
+        titleLabel.alpha = 0
+        
+        let width = view.bounds.width
+        let elements = [emailField, passwordField, loginButton]
+        for element in elements {
+            element?.alpha = 0
+            element?.transform = element!.transform.translatedBy(x: -width, y: 0)
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        UIView.animate(withDuration: 1.5,
+                       delay: 0,
+                       options: [.curveEaseInOut],
+                       animations: { [self] in
+                        titleLabel.alpha = 1
+                        titleLabel.transform = .identity
+                        self.view.layoutIfNeeded()
```

Comment:
You only need to call _view.layoutIfNeeded()_ when animating AutoLayout constraints. You can remove that from here, and you will see it still works.